src/components/TodoInsert.tsx

```diff
@@ -0,0 +1,43 @@
+import { useState } from "react";
+import { InsertButton, InsertForm, StyledInput } from "./TodoInsert.styled.ts";
+import { MdAdd } from "react-icons/md";
+
+interface TodoInsertProps {
+  onAddTodo: (text: string) => void;
+}
+
+function TodoInsert({ onAddTodo }: TodoInsertProps) {
+  const [value, setValue] = useState("");
```

Comment:
value의 타입이 어떤 타입인지, useState의 제네릭을 활용하여 처리할 수 있어요. 

```suggestion
  const [value, setValue] = useState<string>("");
```

하지만 현재는 따로 제네릭으로 타입을 지정해두지 않았음에도 string을 추론하고 있습니다. 
무슨 이유 때문일까요?

Reply:
타입스크립트가 초기값에 해당하는 `""`을 자동으로 string이라 추론하기 때문입니다. 좀 더 찾아보니 ""와 같은 리터럴 문자가 초기값으로 사용될 때는 더 넓은 타입인 String으로 확장되어 추론된다고 합니다. 
하지만 이번에는 운이 좋게도(?) 제가 생각한 초기값("", 공백)과 실제로 들어가야 하는 값(string)이 맞아떨어졌다고 생각합니다. 또 타입스크립트에서 타입을 검사해주고 확장되어 추론이 되는 것은 좋으나 제 코드를 처음 보는 개발자를 고려하여 `<string>` 명시하는 것이 좋을 것 같습니다!